### ISSUE TITLE
Error on https://www.cooljuegos.com/embed/red-driver-4/

### ISSUE THREAD

**Issue by WeiDavit02** [ruffle-rs/ruffle#5874]
# Error Info
Error name: Error
Error message: panicked at 'index out of bounds: the len is 2 but the index is 2', /home/runner/.cargo/git/checkouts/nihav-vp6-660fee9efdd4622a/9416fcc/nihav-duck/src/codecs/vp56.rs:1032:12
Error stack:
```
Error: panicked at 'index out of bounds: the len is 2 but the index is 2', /home/runner/.cargo/git/checkouts/nihav-vp6-660fee9efdd4622a/9416fcc/nihav-duck/src/codecs/vp56.rs:1032:12
    at s.wbg.__wbg_new_df6e6ab7a65c4c4d (https://www.cooljuegos.com/emulators/ruffle/ruffle.js?id=8abf0a0ffc50a9f61348:1:7628)
    at std::sync::once::Once::call_once::{{closure}}::h6f6ea56fd6eb0153 (https://www.cooljuegos.com/emulators/ruffle/8abf0a0ffc50a9f61348.wasm:wasm-function[999]:0x3022e9)
    at std::sync::once::Once::call_inner::h02a6489b0ecfb269 (https://www.cooljuegos.com/emulators/ruffle/8abf0a0ffc50a9f61348.wasm:wasm-function[1359]:0x37fe26)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::he60880e8d4402e2b (https://www.cooljuegos.com/emulators/ruffle/8abf0a0ffc50a9f61348.wasm:wasm-function[4748]:0x50ef4e)
    at std::panicking::rust_panic_with_hook::h364974e717ed59ec (https://www.cooljuegos.com/emulators/ruffle/8abf0a0ffc50a9f61348.wasm:wasm-function[3359]:0x4c7423)
    at rust_begin_unwind (https://www.cooljuegos.com/emulators/ruffle/8abf0a0ffc50a9f61348.wasm:wasm-function[4809]:0x5115ba)
    at core::panicking::panic_fmt::hf69c8b08bc9d2ee5 (https://www.cooljuegos.com/emulators/ruffle/8abf0a0ffc50a9f61348.wasm:wasm-function[5017]:0x51c179)
    at core::panicking::panic_bounds_check::h1d84519ec46ed48c (https://www.cooljuegos.com/emulators/ruffle/8abf0a0ffc50a9f61348.wasm:wasm-function[4222]:0x4fdd4f)
    at nihav_duck::codecs::vp56::VP56Decoder::predict_dc::h984e4648691c27a4 (https://www.cooljuegos.com/emulators/ruffle/8abf0a0ffc50a9f61348.wasm:wasm-function[1225]:0x356eff)
    at nihav_duck::codecs::vp56::VP56Decoder::decode_planes::hbf2312b282c38c4f (https://www.cooljuegos.com/emulators/ruffle/8abf0a0ffc50a9f61348.wasm:wasm-function[311]:0x6bfd0)
```

# Player Info
Allows script access: false

# Page Info
Page URL: https://www.cooljuegos.com/embed/red-driver-4/
SWF URL: https://data.gameflare.com/games/5423/jfJeSnkokqevRm.swf

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.93 Safari/537.36
OS: Win32

# Ruffle Info
Version: 0.1.0
Name: nightly 2021-12-02
Channel: nightly
Built: 2021-12-02T00:11:23.895Z
Commit: e3dfeb5a44c3a728cfe50642e480f3ea3498c099
Is extension: false

**Comment by torokati44** [comment]
Thanks for the report!
Do you perhaps remember what you did, or what was happening in the game, in the moment the error happened?
Any small detail can matter...
Or can you perhaps reproduce the error on purpose? That would help us a lot in finding and fixing the cause of this crash.

**Comment by torokati44** [comment]
Anyway, this is most likely the same as https://github.com/ruffle-rs/ruffle/issues/5911 (which I was able to reproduce), given that the same error happens at the same location in the code, and the two games are very similar, using the same techniques to achieve some of their visual effects.

**Comment by torokati44** [comment]
This too should be fixed by #5936.

**Comment by WeiDavit02** [comment]
The problem is that the game its too old, and some thinks doesnt even work,
like when you crash, there are no cars, i sugest that maybe recode the game.
